src/index.ts

```diff
@@ -76,13 +80,13 @@ export class Iterable<T> implements LibIterable<T> {
         }
 
         const _that = this;
-        return new Iterable<T>((function* filterGenerator() {
+        return new Iterable<T>(function* () {
```

Comment:
I ran some benchmarks on moving this generator to a private method and didn't see any measurable improvements. I left it inline since it's more readable